src/compile/codegen.rs

```diff
@@ -1633,7 +1633,7 @@ fn mavm_codegen_expr<'a>(
                 vec![Type::Uint, Type::Any],
                 Box::new(array_type.clone()),
             );
-            let default_val = base_type.default_value();
+            let default_val = base_type.default_value().unwrap_or(Value::none());
```

Comment:
As currently written it looks like we always change a `None` return value to `Value::none()`, if we don't expect to change this soon it would be simpler to simply return `Value::none()` directly, and change the signature of `default_value` from `Option<Value>` back to `Value`

Reply:
Revised based on out-of-band conversation.